theseus/utils/examples/bundle_adjustment/reprojection_error.py

```diff
@@ -0,0 +1,123 @@
+from typing import List, Optional, Tuple
+
+import torch
+
+import theseus as th
+
+from .util import soft_loss_huber_like
+
+
+class ReprojectionError(th.CostFunction):
+    def __init__(
+        self,
+        camera_pose: th.SE3,
+        log_loss_radius: th.Vector,
+        world_point: th.Point3,
+        image_feature_point: th.Point2,
+        focal_length: th.Vector,
```

Comment:
nit: for ease in reading maybe order them based on hoe they are registered, optim first then aux?

Reply:
Done!